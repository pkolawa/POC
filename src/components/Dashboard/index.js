import React from "react";
import { withStyles } from "@material-ui/core/styles";
import Container from "@material-ui/core/Container";
import Typography from "@material-ui/core/Typography";
import Paper from "@material-ui/core/Paper";
import { Link } from "react-router-dom";

const styles = (theme) => ({
  root: {
    marginTop: '30px',
    display: "flex",
    flexWrap: "wrap",
    justifyContent: "center",
    "& > *": {
      margin: theme.spacing(1),
      width: theme.spacing(16),
      height: theme.spacing(16),
    },
    "& a": {
      textDecoration: "none",
      display: "inline-flex",
      alignItems: "center",
      height: "inherit",
      width: "inherit",
      justifyContent: "center"
    }
  },
});

class Dashboard extends React.Component {
  render() {
    const { classes } = this.props;
    return (
      <React.Fragment>
        <Container maxWidth="sm">
          <div className={classes.root}>
            <Paper elevation={3}>
              <Link to="/configuration">
                <Typography align="center" color="primary">
                  Configuration Page
                </Typography>
              </Link>
            </Paper>
            <Paper elevation={3}>
              <Link to="/monitor">
                <Typography align="center" color="primary">
                  Monitor Page
                </Typography>
              </Link>
            </Paper>
            <Paper elevation={3}>
              <Link to="/runtime">
                <Typography align="center" color="primary">
                  Runtime Page
                </Typography>
              </Link>
            </Paper>
          </div>
        </Container>
      </React.Fragment>
    );
  }
}

export default withStyles(styles)(Dashboard);
