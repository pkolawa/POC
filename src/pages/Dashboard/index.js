import React from "react";
import { Helmet } from "react-helmet";
import DashboardContainer from "../../containers/DashboardContainer";

class Dashboard extends React.Component {
  constructor(props) {
    super(props);
  }
  render() {
    return (
      <>
        <Helmet>
          <title>Dashboard</title>
          <meta
            name="description"
            content="Dashboard"
          />
        </Helmet>
        <DashboardContainer />
      </>
    );
  }
}

export default Dashboard;
