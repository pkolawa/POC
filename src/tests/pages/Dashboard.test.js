import React from "react";
import { shallow } from "enzyme";
import Dashboard from "../../pages/Dashboard";

test("Should Render Dashboard Page", () => {
  const wrapper = shallow(<Dashboard/>);
  expect(wrapper).toMatchSnapshot();
});