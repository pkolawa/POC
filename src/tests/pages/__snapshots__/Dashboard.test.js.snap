// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should Render Dashboard Page 1`] = `
<Fragment>
  <HelmetWrapper
    defer={true}
    encodeSpecialCharacters={true}
  >
    <title>
      Dashboard
    </title>
    <meta
      content="Dashboard"
      name="description"
    />
  </HelmetWrapper>
  <Connect(WithStyles(DashboardContainer)) />
</Fragment>
`;
